discovery-infra/test_infra/helper_classes/kube_helpers.py

```diff
@@ -0,0 +1,376 @@
+
+
+class InstallStrategy(dict):
+
+    def __init__(
+            self,
+            control_plane_agents: int = 1,
+            worker_agents: int = 0,
+            label_selector: Optional[Dict[str, str]] = None,
+            host_prefix: int = env_variables['host_prefix'],
+            machine_cidr: str = DEFAULT_MACHINE_CIDR,
+            cluster_cidr: str = DEFAULT_CLUSTER_CIDR,
+            service_cidr: str = DEFAULT_SERVICE_CIDR,
+            ssh_public_key: str = env_variables['ssh_public_key'],
+    ):
+        dict.__init__(
+            self,
+            agent={
+                'networking': {
+                    'clusterNetwork': [{
+                        'cidr': cluster_cidr,
+                        'hostPrefix': host_prefix
+                    }],
+                    'serviceNetwork': [service_cidr]
+                },
+                'provisionRequirements': {
+                    'controlPlaneAgents': control_plane_agents,
+                    'workerAgents': worker_agents,
+                },
+                'agentSelector': {'matchLabels': label_selector or {}}
+            }
+        )
+
+        if _does_string_contain_value(ssh_public_key):
+            self['agent']['sshPublicKey'] = ssh_public_key
+
+        if _does_string_contain_value(machine_cidr):
+            self['agent']['machineNetwork'] = [{'cidr': machine_cidr}],
+
+
+DEFAULT_WAIT_FOR_CRD_STATUS_TIMEOUT = 60
+DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT = 300
+
+
+class ClusterDeployment(_BaseKubeAPIResource):
+    hive_api_group = 'hive.openshift.io'
+
+    def __init__(self, name: str, namespace: str = env_variables['namespace']):
+        _BaseKubeAPIResource.__init__(self, name, namespace)
+
+        self.crd_api = CustomObjectsApi(self.api_client)
+
+        self.__secret = None
+
+    @property
+    def secret(self):
+        return self.__secret
+
+    def create(
+            self,
+            platform: Platform,
+            install_strategy: InstallStrategy,
+            secret: Secret,
+            base_domain: str = env_variables['base_domain'],
+            **kwargs
+    ) -> None:
+
+        body = {
+            'apiVersion': f'{self.hive_api_group}/v1',
+            'kind': 'ClusterDeployment',
+            'metadata': self.ref,
+            'spec': {
+                'clusterName': self.ref['name'],
+                'baseDomain': base_domain,
+                'platform': platform,
+                'provisioning': {'installStrategy': install_strategy},
+                'pullSecretRef': secret.ref,
+            }
+        }
+        body['spec'].update(kwargs)
+        self.crd_api.create_namespaced_custom_object(
+            group=self.hive_api_group,
+            version='v1',
+            plural='clusterdeployments',
+            body=body,
+            namespace=self.ref['namespace']
+        )
+        self.__secret = secret
+
+        logger.info('created cluster deployment %s: %s', self.ref, body)
```

Comment:
is the body a large block of text?
consider using ``pprint`` if so, otherwise it's ok

Reply:
good idea, 10x